src/main/java/io/strimzi/kafka/bridge/http/HttpSinkBridgeEndpoint.java

```diff
@@ -462,13 +470,15 @@ public void doListSubscriptions(RoutingContext routingContext) {
      * @param routingContext the routing context
      */
     public void doUnsubscribe(RoutingContext routingContext) {
-        this.unsubscribe(unsubscribeResult -> {
-            if (unsubscribeResult.succeeded()) {
+        // fulfilling the request in a separate thread to free the Vert.x event loop still in place
+        CompletableFuture.runAsync(() -> this.unsubscribe()).whenComplete((v, ex) -> {
```

Comment:
Am I alone in finding these lines with both `runAsync()` and `whenComplete()` to be not as clear as putting them on separate lines?

Reply:
I did just because the kind of "very few" code in the runAsync, but I am fine to move it on next line ;-)